File: backend/gd-roadmap-service/src/main/java/org/yc/gnosdrasil/gdroadmapservice/entities/Roadmap.java
package org.yc.gnosdrasil.gdroadmapservice.entities;

import lombok.*;
import org.springframework.data.neo4j.core.schema.Id;
import org.springframework.data.neo4j.core.schema.Node;
import org.springframework.data.neo4j.core.schema.Relationship;

import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Node
public class Roadmap {
    @Id
    public String title;

    @Relationship(type = "OF", direction = Relationship.Direction.INCOMING)
    public List<Step> steps;
}

File: frontend/src/types/roadmap.ts
export interface ConnectedStep {
  id: string;
  label: string;
  connectedSteps: ConnectedStep[];
}

export interface Step {
  id: string;
  label: string;
  connectedSteps: ConnectedStep[];
}

export interface Roadmap {
  id: string | null;
  title: string;
  steps: Step[];
} 